contrib/p5_wild_wyss/src/logger/logger.js

```diff
@@ -155,62 +156,88 @@ const LOG_NOTHING = Pair(n9)("NOTHING");
 
 /**
  * Creates a new logger at log level {@link LOG_DEBUG}.
  * @example
- * const debug = debugLogger(() => [Appender()])("ch.fhnw")(_context => _level => id);
+ * const debug = debugLogger("ch.fhnw")(_context => _level => id);
  * debug("a message to log to console");
  * // writes "a message to log to console" to the console
  */
-const debugLogger = () =>
-    logger(LOG_DEBUG)(appenderList.map(app => app.debug));
+const debugLogger =  logger(LOG_DEBUG);
 
 /**
  * Creates a new logger at log level {@link LOG_INFO}.
  * @example
- * const debug = debugLogger(() => [Appender()])("ch.fhnw")(_context => _level => id);
+ * const debug = infoLogger("ch.fhnw")(_context => _level => id);
  * debug("a message to log to console");
  * // writes "a message to log to console" to the console
  */
-const infoLogger = () =>
-    logger(LOG_INFO)(appenderList.map(app => app.info));
+const infoLogger = logger(LOG_INFO);
 
 /**
  * Creates a new logger at log level {@link LOG_WARN}.
  * @example
- * const warn = warnLogger(() => [Appender()])("ch.fhnw")(_context => _level => id);
+ * const warn = warnLogger("ch.fhnw")(_context => _level => id);
  * warn("a message to log to console");
  * // writes "a message to log to console" to the console
  */
-const warnLogger = () =>
-    logger(LOG_WARN)(appenderList.map(app => app.warn));
+const warnLogger = logger(LOG_WARN);
 
 /**
  * Creates a new logger at log level {@link LOG_ERROR}.
  * @example
- * const error = errorLogger(() => [Appender()])("ch.fhnw")(_context => _level => id);
+ * const error = errorLogger("ch.fhnw")(_context => _level => id);
  * error("a message to log to console");
  * // writes "a message to log to console" to the console
  */
-const errorLogger = () =>
-    logger(LOG_ERROR)(appenderList.map(app => app.error));
+const errorLogger = logger(LOG_ERROR);
 
 /**
  * Creates a new logger at log level {@link LOG_FATAL}.
  * @example
- * const fatal = fatalLogger(() => [Appender()])("ch.fhnw")(_context => _level => id);
+ * const fatal = fatalLogger("ch.fhnw")(_context => _level => id);
  * fatal("a message to log to console");
  * // writes "a message to log to console" to the console
  */
-const fatalLogger = () =>
-    logger(LOG_FATAL)(appenderList.map(app => app.fatal));
+const fatalLogger = logger(LOG_FATAL);
+
+/**
+ * This is a state.
+ * The currently active {@link AppenderType AppenderType's}.
+ * @type { AppenderType[] }
+ */
+const appenderList = [];
+
+/**
+ * Adds one or multiple {@link AppenderType AppenderType's} to the appender list.
+ * @param newAppender
+ */
+const addToAppenderList = (...newAppender) => newAppender.forEach(app => appenderList.push(app));
+
+/**
+ * Removes a given {@link AppenderType} from the current appender list.
+ *
+ * @param   { AppenderType   } item
+ * @returns { AppenderType[] }
+ */
+const removeFromAppenderList = item => {
+  // correct type is not recognized here.
+  return /** @type { AppenderType[] }*/ removeItem(appenderList)(item);
```

Comment:
we should return a copy of the appenderlist here using `[...removeItem(appenderList)(item)]`